src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs

```diff
@@ -2671,8 +2671,14 @@ public static void IsIanaIdTest()
                 Assert.True((expected || tzi.Id.Equals("Utc", StringComparison.OrdinalIgnoreCase)) == tzi.HasIanaId, $"`{tzi.Id}` has wrong IANA Id indicator");
             }
 
-            string timeZoneIdNotIANAId = PlatformDetection.IsiOS || PlatformDetection.IstvOS ? "America/Buenos_Aires" : "Pacific Standard Time";
-            Assert.False(TimeZoneInfo.FindSystemTimeZoneById(timeZoneIdNotIANAId).HasIanaId, $" should not be IANA Id.");
+            if (!PlatformDetection.IsiOS && !PlatformDetection.IstvOS)
+            {
+                Assert.False(TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time").HasIanaId, $" should not be IANA Id.");
+            }
+            else
+            {
+                Assert.Throws<TimeZoneNotFoundException>(() => TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time").HasIanaId);
```

Comment:
>HasIanaId [](http://example.com/codeflow?start=124&length=9)

nit: Maybe you don't need calling this property here? it wouldn't hurt if you keep it though.

Reply:
I was thinking similarly since it is unnecessary, but was wondering if keeping the `HasIanaId` would be in the spirit of the test purpose of checking if it was Iana Id.